tests/test_opentime.cpp

```diff
@@ -0,0 +1,39 @@
+#include "tests.h"
+
+#include <opentime/rationalTime.h>
+
+namespace otime = opentime::OPENTIME_VERSION;
+
+int
+main(int argc, char** argv) {
+    Tests tests;
+
+    tests.add_test("test_create", [] {
+        double              t_val = 30.2;
+        otime::RationalTime t(t_val);
+        assertEqual(t.value(), t_val);
+
+        t = otime::RationalTime();
+        assertEqual(t.value(), 0.0);
```

Comment:
Could we add an epsilon to this assert, to compare floating points?
In the C tests I use a value (suggested by Nick) of 5.82077e-11. Reason as mentioned in the [C test](https://github.com/OpenTimelineIO/OpenTimelineIO-C-Bindings/blob/main/tests/epsilon.h):
```
/**
 * We are not testing values outside of one million seconds.
 * At one million second, and double precision, the smallest
 * resolvable number that can be added to one million and return
 * a new value one million + epsilon is 5.82077e-11.
 *
 * This was calculated by searching iteratively for epsilon
 * around 1,000,000, with epsilon starting from 1 and halved
 * at every iteration, until epsilon when added to 1,000,000
 * resulted in 1,000,000.
 */
```

Also, I could help writing a few tests if you'd like.

Reply:
Good suggestion about the epsilon, I'll add that to specialized versions of ```assertEqual()``` and ```assertNotEqual()``` for double types.

Helping out with tests would be great, thanks! I just added a couple of tests to get feedback about the style first.

One thing I was wondering, does there need to be a single ```main()``` function for the tests to get the code coverage to work?

Reply:
OK, I added the epsilon for double comparison, check it out and see what you think. I hope you don't mind, I copied your comment about the epsilon value.

Reply:
This looks good!

W.r.t single main(), I'm not sure because I don't have experience using coverage with C/CPP. I looked at OCIO. Correct me if I'm wrong, I think they are using a single main().

https://github.com/AcademySoftwareFoundation/OpenColorIO/tree/main/tests/cpu

Reply:
It looks like the coverage accumulates, so a single main() is not needed:
"The execution counts are cumulative. If the example program were executed again without removing the .gcda file, the count for the number of times each line in the source was executed would be added to the results of the previous run(s). This is potentially useful in several ways. For example, it could be used to accumulate data over a number of program runs as part of a test verification suite, or to provide more accurate long-term information over a large number of program runs."
https://linux.die.net/man/1/gcov